bundles/framework/layerlist/view/LayerViewTabs/LayerList/LayerCollapse/LayerCollapsePanel.jsx

```diff
@@ -23,71 +43,203 @@ const StyledListItem = styled(ListItem)`
 `;
 
 const StyledEditGroup = styled.span`
-    padding-right: 20px;
-    padding-bottom: 1px;
+    padding-right: 5px;
 `;
 
 const renderLayer = ({ model, even, selected, controller }) => {
     const itemProps = { model, even, selected, controller };
     return (
-        <StyledListItem>
-            <Layer key={model.getId()} {...itemProps} />
-        </StyledListItem>
+            <StyledListItem>
+                <Layer key={model.getId()}  {...itemProps} />
+            </StyledListItem>
     );
 };
+
 renderLayer.propTypes = {
     model: PropTypes.any,
     even: PropTypes.any,
     selected: PropTypes.any,
     controller: PropTypes.any
 };
 
-const onToolClick = (event, id, groupMethod, tool, layerCountInGroup) => {
+const onToolClick = (event, tool, group) => {
+    const id = group.getId();
+    const parentId = group.getParentId();
+    const groupMethod = group.getGroupMethod();
+    const layerCountInGroup = group.getLayers().length;
     const cb = tool.getCallback();
     if (cb) {
-        cb(event, id, groupMethod, layerCountInGroup);
+        cb(event, id, groupMethod, layerCountInGroup, parentId);
     }
     // Prevent collapse open on tool icon click
     event.stopPropagation();
 };
 
+const selectGroup = (event, setVisible, checked, group, controller) => {
+    setVisible(false);
+    // if switch is checked, we add the groups layers to selected layers, if not, we remove all the layers from checked layers
+    !checked ? controller.addGroupLayersToMap(group) : controller.removeGroupLayersFromMap(group); 
+    event.stopPropagation();
+}
+
+const onGroupSelect = (event, setVisible, checked, group, controller) => { 
+    // check if we need to show warning (over 10 layers inside the group)
+    if(checked && group.layers.length > 1) {
+        setVisible(true);
+    } else {
+        selectGroup(event, setVisible, !checked, group, controller);
+    }
+    event.stopPropagation();
+};
+
+const onCancel = (event, setVisible) => {
+    setVisible(false);
+    event.stopPropagation();
+}
+
+
+const SubCollapsePanel = ({ active, group, selectedLayerIds, selectedGroupIds, controller, propsNeededForPanel }) => {
```

Comment:
I don't find where `selectedGroupIds` is used. Can it be removed? Also I don't see a reason for passing `active` here since selectedLayerIds and group that are used to calculate the value is passed here. So move the logic of calculating "active" here instead and remove it from props. What is in `propsNeededForPanel`? Is it needed or can it be removed as well?

Reply:
Active is needed as it defines if toggle is on based on if all layers are selected. Others have now been cleaned up.